pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -354,12 +351,22 @@ func (asw *actualStateOfWorld) MarkVolumeAsUnmounted(
 
 func (asw *actualStateOfWorld) MarkDeviceAsMounted(
 	volumeName v1.UniqueVolumeName, devicePath, deviceMountPath string) error {
-	return asw.SetVolumeGloballyMounted(volumeName, true /* globallyMounted */, devicePath, deviceMountPath)
+	return asw.SetDeviceMountState(volumeName, operationexecutor.DeviceGloballyMounted, devicePath, deviceMountPath)
+}
+
+func (asw *actualStateOfWorld) MarkDeviceAsUncertain(
+	volumeName v1.UniqueVolumeName, devicePath, deviceMountPath string) error {
+	return asw.SetDeviceMountState(volumeName, operationexecutor.DeviceMountUncertain, devicePath, deviceMountPath)
+}
+
+func (asw *actualStateOfWorld) MarkVolumeMountAsUncertain(markVolumeOpts operationexecutor.MarkVolumeMountedOpts) error {
+	markVolumeOpts.VolumeMountState = operationexecutor.VolumeMountUncertain
```

Comment:
This line https://github.com/kubernetes/kubernetes/pull/82492/files#diff-450e811a4953f760ff1594ede8b2037eR753 already set the state to uncertain. Why need to set it up here too?

Reply:
My thinking is - this is to safeguard against callers who may not be setting this correctly. The function is called `MarkVolumeMountAsUncertain`, so caller may expect that volume will be marked uncertain without explicitly specifying the state.